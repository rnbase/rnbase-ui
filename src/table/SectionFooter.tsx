import React from 'react';
import { StyleProp, StyleSheet, Text, TextStyle, View } from 'react-native';

import { Theme, Themed, withTheme } from '../theming';

interface SectionFooterProps {
  style?: StyleProp<TextStyle>;
}

const SectionFooter: React.FC<Themed<typeof createStyleSheet, SectionFooterProps>> = ({
  theme: { styles },
  children,
  style,
}) => {
  if (!children) {
    return null;
  }

  if (React.isValidElement(children)) {
    return <View style={[styles.root, style]}>{children}</View>;
  }

  return (
    <View style={styles.root}>
      <Text style={[styles.text, style]}>{children}</Text>
    </View>
  );
};

const createStyleSheet = ({ colors, fonts }: Theme) =>
  StyleSheet.create({
    root: {
      marginTop: 7.5,
      marginHorizontal: 20,
    },
    text: {
      ...fonts.normal,
      fontSize: 13,
      color: colors.gray,
    },
  });

export default withTheme(SectionFooter, createStyleSheet, 'Table.SectionFooter');
