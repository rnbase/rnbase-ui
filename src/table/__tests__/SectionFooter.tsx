import React from 'react';

// Note: test renderer must be required after react-native.
import TestRenderer from 'react-test-renderer';

/**
 * Under test
 */
import SectionFooter from '../SectionFooter';

type Props = React.ComponentProps<typeof SectionFooter>;

const Component = (props: any) => React.createElement('Component', props);

const createElement = (props: Partial<Props>) => <SectionFooter {...props} />;

const createRenderer = (props: Partial<Props>) => TestRenderer.create(createElement(props));

it('should render normally as text', () => {
  expect(createRenderer({ children: 'TEXT' })).toMatchSnapshot();
});

it('should render normally as component', () => {
  expect(createRenderer({ children: <Component /> })).toMatchSnapshot();
});

it('should render normally as text with custom styles', () => {
  expect(createRenderer({ children: 'TEXT', style: { color: 'red' } })).toMatchSnapshot();
});

it('should render normally as component with custom styles', () => {
  expect(createRenderer({ children: <Component />, style: { color: 'red' } })).toMatchSnapshot();
});
