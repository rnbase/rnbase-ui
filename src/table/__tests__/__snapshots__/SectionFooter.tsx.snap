// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render normally as component 1`] = `
<View
  style={
    Array [
      Object {
        "marginHorizontal": 20,
        "marginTop": 7.5,
      },
      undefined,
    ]
  }
>
  <Component />
</View>
`;

exports[`should render normally as component with custom styles 1`] = `
<View
  style={
    Array [
      Object {
        "marginHorizontal": 20,
        "marginTop": 7.5,
      },
      Object {
        "color": "red",
      },
    ]
  }
>
  <Component />
</View>
`;

exports[`should render normally as text 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 20,
      "marginTop": 7.5,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "#838387",
          "fontFamily": "System",
          "fontSize": 13,
          "fontWeight": "400",
          "includeFontPadding": false,
          "textAlignVertical": "center",
        },
        undefined,
      ]
    }
  >
    TEXT
  </Text>
</View>
`;

exports[`should render normally as text with custom styles 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 20,
      "marginTop": 7.5,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "#838387",
          "fontFamily": "System",
          "fontSize": 13,
          "fontWeight": "400",
          "includeFontPadding": false,
          "textAlignVertical": "center",
        },
        Object {
          "color": "red",
        },
      ]
    }
  >
    TEXT
  </Text>
</View>
`;
